walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,127 @@
 // limitations under the License.
 
 package com.google.sps;
-
+ 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
+  private static int FULL_DAY = 24 * 60;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    ArrayList<TimeRange> availableTimeRanges = getAvailableTimeRanges(events, request, request.getDuration());
+    return availableTimeRanges;
+
+  }
+
+  // returns what events have attendees going to the requested event
+  public ArrayList<Event> checkMatchingAttendees(Collection<Event> events, MeetingRequest request) {
```

Comment:
nit: The naming of this method is a bit confusing. I suggest getEventsWithMatchingAttendees(...) or something similar.